Python/Product/TestAdapter.Executor/UnitTest/TestExecutorUnitTest.cs

```diff
@@ -530,8 +531,9 @@ sealed class TestRunner : IDisposable {
                         // If there's an error in the launcher script,
                         // it will terminate without connecting back.
                         WaitHandle.WaitAny(new WaitHandle[] { _connected, proc.WaitHandle });
+                        bool processConnected = _connected.WaitOne(1);
```

Comment:
I wonder if proc.WaitHandle was null sometimes causing the WaitAny to exit too quickly.

Reply:
It was going inside the if statement which checks for `ExitCode`, so it wasn't returning from `WaitAny` too quickly, it was just really already done executing the script.

Reply:
I just tested what happens when you pass `null` for one of the array elements to `WaitAny`, and it throws an `ArgumentNullException`